### PR TITLE
minimal fixes to run Phase2 patatrack pixel only workflows without using phase-1 pixel conditions

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Minimal fix for https://github.com/cms-sw/cmssw/pull/42794#issuecomment-1727027260

#### PR validation:

`runTheMatrix.py --what upgrade -l 24834.5,20834.501,20834.502 -t 4 -j 8` runs fine.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

### CODE REVIEW

File: RecoLocalTracker/SiPixelClusterizer/python/SiPixelClusterizerPreSplitting_cfi.py [hunk 1]
```diff
@@ -10,9 +10,10 @@
     cpu = _siPixelClusters.clone()
 )
 
+from Configuration.Eras.Modifier_phase2_tracker_cff import phase2_tracker
 from Configuration.ProcessModifiers.gpu_cff import gpu
 # SwitchProducer wrapping the legacy pixel cluster producer or an alias for the pixel clusters information converted from SoA
-gpu.toModify(siPixelClustersPreSplitting,
+(gpu & ~phase2_tracker).toModify(siPixelClustersPreSplitting,
```
Comment (mmusich) [inline RecoLocalTracker/SiPixelClusterizer/python/SiPixelClusterizerPreSplitting_cfi.py:16]: here probably we don't want to remove the modifier for the cuda part... not sure how to do that though

**Comment by mmusich** [comment]
enable gpu

File: RecoLocalTracker/SiPixelClusterizer/python/SiPixelClusterizerPreSplitting_cfi.py (hunk 1 above)
Comment (kpedro88) [reply RecoLocalTracker/SiPixelClusterizer/python/SiPixelClusterizerPreSplitting_cfi.py:16]: just split it into two `toModify` commands?

**Comment by mmusich** [comment]
test parameters:
   * enable_tests = gpu
   * workflows_gpu = 20834.501, 20834.502
   * workflows = 24834.5
   * relvals_opt= -w upgrade

**Comment by mmusich** [comment]
@cmsbuild, please test

File: RecoLocalTracker/SiPixelClusterizer/python/SiPixelClusterizerPreSplitting_cfi.py (hunk 1 above)
Comment (mmusich) [reply RecoLocalTracker/SiPixelClusterizer/python/SiPixelClusterizerPreSplitting_cfi.py:18]: done in the last push.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
> ValueError: Undefined workflows: 20834.501, 20834.502

Ugh....

**Comment by mmusich** [comment]
test parameters:
   * enable_tests = gpu
   * workflows_gpu =  20834.501, 20834.502
   * workflows = 24834.5
   * relvals_opt = -w upgrade,gpu,standard
   * relvals_opt_gpu = -w upgrade,gpu,standard

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by jfernan2** [comment]
+1
This fix is fine from RECO point of view

**Comment by mmusich** [comment]
@cms-sw/orp-l2 this PR should fix the remaining [IB relval failures](https://cmssdt.cern.ch/SDT/html/cmssdt-ib/#/relVal/CMSSW_13_3/2023-09-20-2300?selectedArchs=el9_amd64_gcc11&selectedFlavors=X&selectedStatus=failed), please take a look.

**Comment by iarspider** [comment]
@cms-sw/operations-l2 @cms-sw/orp-l2 gentle ping.

**Comment by antoniovilela** [comment]
+1

**Pull request merged**